rudder-core/src/main/scala/com/normation/rudder/services/reports/NodeChangesService.scala

```diff
@@ -59,111 +60,105 @@ import net.liftweb.http.js.JE._
+   *
    */
-  def getCurrentValidIntervals(since: Option[DateTime]): Seq[Interval]
+  final def getCurrentValidIntervals(since: Option[DateTime]): List[Interval] = {
+    val startTime = since.getOrElse(DateTime.now.minusDays(changesMaxAge))
+    val endTime   = DateTime.now
+    getInterval(startTime, endTime)
+  }
 
   /*
-   * The two following methods are pure algorithm methods and we want that all
+   * The following method is pure algorithm methods and we want that all
    * implementation of getChangesByInterval use them.
    */
 
   /**
-   * Group the given set of reports by rules and based on the allowed intervals
-   * given in parameter. All report outside any of the intervals is discared.
-   */
-  def filterGroupByInterval(intervals: Seq[Interval], changes: Seq[ResultRepairedReport]): ChangesByRule = {
-    changes.groupBy(_.ruleId).mapValues { ch =>
-      ( for {
-         interval <- intervals
-      } yield {
-        (interval,ch.filter(interval contains _.executionTimestamp))
-      } ).toMap
-    }
-  }
-
-  /**
    * Stable computation of intervals of 6h starting at 0,6,12 or 18
    * between two dates.
    *
    * The date are always included in the resulting interval
    * sequence, which is also ordered by date.
    */
-  def getInterval(since: DateTime, to: DateTime): Seq[Interval] = {
-    // We want our first interval limit to be 00:00 | 06:00 | 12:00 | 18:00
-
-    val firstInterval = {
-      val firstHour =
-        if ( to isBefore to.withHourOfDay(6) ) {
-          to.withTimeAtStartOfDay()
-        } else {
-          if (to isBefore to.withHourOfDay(12)) {
-            to.withTimeAtStartOfDay().withHourOfDay(6)
-          } else {
-            if (to isBefore to.withHourOfDay(18)) {
-              to.withTimeAtStartOfDay().withHourOfDay(12)
-            } else {
-              to.withTimeAtStartOfDay().withHourOfDay(18)
-        } } }
-      // Fix minutes to 0
-      new Interval(firstHour, firstHour.plusHours(6))
-    }
 
-    // Compute intervals of changes recursively from the last interval lower
-    def computeIntervals(previousLowerBound: DateTime): List[Interval] = {
-      if ( previousLowerBound isBefore since) {
-        // Our last bound is before the limit, do not create a interval
-        Nil
-      } else {
-        val nextLowerBound = previousLowerBound.minusHours(6)
-        val interval = new Interval(nextLowerBound,previousLowerBound)
-        interval :: computeIntervals(nextLowerBound)
+  private[this] final def getInterval(since: DateTime, to: DateTime): List[Interval] = {
+    if(to.isBefore(since)) {
+      Nil
+    } else {
+      // We want our first interval limit to be 00:00 | 06:00 | 12:00 | 18:00
+
+      //utility that create an interval from the given date to date+6hours
+      def sixHours(t: DateTime): Interval = {
+        //6 hours in milliseconds
+        new Interval(t, new DateTime(t.getMillis + 6l * 3600 * 1000))
```

Comment:
Police in English refers to the state-funded folk that keep order. I think you're talking about a font :)